assets/app/view/game/tile.rb

```diff
@@ -25,13 +25,16 @@ def render_tile_part(part_class, **kwargs)
       def should_render_revenue?
         revenue = @tile.revenue_to_render
 
+        # special case: city with multi-revenue - no choice but to draw separate revenue
+        return true if revenue.any? { |r| !r.is_a?(Numeric) }
+
         return false if revenue.empty?
 
         return false if revenue.first.is_a?(Numeric) && (@tile.cities + @tile.towns).one?
 
         return false if revenue.uniq.size > 1
 
-        return false if @tile.cities.sum(&:slots) < 3 && @tile.stops.size == 2
+        return false if @tile.cities.sum(&:slots) < 3 && (@tile.cities + @tile.towns).size == 2
```

Comment:
we call cities + towns . size a lot, maybe make a helper method on tiles

Reply:
Yeah. I originally thought @tile.stops would work, but it only contains cities and town that are connected.

Reply:
well that should be fixed then, that sounds like a bug